internal/controller/postgrescluster/util.go

```diff
@@ -285,3 +287,61 @@ func safeHash32(content func(w io.Writer) error) (string, error) {
 	}
 	return rand.SafeEncodeString(fmt.Sprint(hash.Sum32())), nil
 }
+
+// AdditionalVolumeMount returns the name and mount path of the additional volume.
+func AdditionalVolumeMount(volumeName, claimName string) corev1.VolumeMount {
+	return corev1.VolumeMount{
+		Name:      claimName,
+		MountPath: "/volumes" + "/" + volumeName}
+}
+
+// addAdditionalVolumesToSpecifiedContainers adds additional volumes to the specified
+// containers in the specified pod
+func addAdditionalVolumesToSpecifiedContainers(template *corev1.PodTemplateSpec,
+	additionalVolumes []*v1beta1.AdditionalVolume) {
+
+	for _, additionalVolumeRequest := range additionalVolumes {
+
+		additionalVolumeMount := AdditionalVolumeMount(
+			additionalVolumeRequest.Name,
+			additionalVolumeRequest.ClaimName)
+
+		additionalVolume := corev1.Volume{
+			Name: additionalVolumeMount.Name,
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					// TODO: Using the claim name as the claim name here means this has to be unique
+					// We could add the index if we want someone to mount the same PVC in multiple dirs?
+					ClaimName: additionalVolumeMount.Name,
+					ReadOnly:  additionalVolumeRequest.ReadOnly,
+				},
+			},
+		}
+
+		for i := range template.Spec.Containers {
+			if len(additionalVolumeRequest.Containers) == 0 ||
+				slices.Contains(
+					additionalVolumeRequest.Containers,
+					template.Spec.Containers[i].Name) {
```

Comment:
🔧 Rather than slices.Contains, we can use a [Set.Has](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#Set.Has) and share it with both loops.

```go
names := sets.New(additionalVolumeRequest.Containers...)

for ..Containers {
  if names.Len() == 0 || names.Has(..Name) { .... }
}

for ..InitContainers { .... }
```